format/avro/decoders/record.go

```diff
@@ -0,0 +1,47 @@
+package decoders
+
+import (
+	"fmt"
+
+	"github.com/wader/fq/format/avro/schema"
+	"github.com/wader/fq/pkg/decode"
+)
+
+func decodeRecordFn(schema schema.SimplifiedSchema) (func(string, *decode.D), error) {
+	if len(schema.Fields) == 0 {
+		return nil, fmt.Errorf("record must have fields")
+	}
+	var fieldNames []string
+	var fieldDecoders []func(string, *decode.D)
+
+	for _, f := range schema.Fields {
+		fieldNames = append(fieldNames, f.Name)
+		fc, err := DecodeFnForSchema(f.Type)
+		if err != nil {
+			return nil, fmt.Errorf("failed parsing record field %s: %w", f.Name, err)
+		}
+		fieldDecoders = append(fieldDecoders, fc)
+	}
+
+	//A record is encoded by encoding the values of its fields in the order that they are declared. In other words, a
+	//record is encoded as just the concatenation of the encodings of its fields. Field values are encoded per their
+	//schema. For example, the record schema
+	//	      { "type": "record",
+	//	        "name": "test",
+	//	        "fields" : [ {"name": "a", "type": "long"},
+	//	      				{"name": "b", "type": "string"} ]
```

Comment:
json indent in comment looks a bit weird (see gitbhub diff), i guess it has tab characters, maybe convert to space?